script/tool/test/build_examples_command_test.dart

```diff
@@ -36,16 +39,49 @@ void main() {
     });
 
     test('fails if no plaform flags are passed', () async {
+      Error? commandError;
+      final List<String> output = await runCapturingPrint(
+          runner, <String>['build-examples'], errorHandler: (Error e) {
+        commandError = e;
+      });
+
+      expect(commandError, isA<ToolExit>());
       expect(
-        () => runCapturingPrint(runner, <String>['build-examples']),
-        throwsA(isA<ToolExit>()),
-      );
+          output,
+          containsAllInOrder(<Matcher>[
+            contains('At least one platform must be provided'),
+          ]));
+    });
+
+    test('fails if building fails', () async {
+      createFakePlugin('plugin', packagesDir,
+          platformSupport: <String, PlatformSupport>{
+            kPlatformIos: PlatformSupport.inline
+          });
+
+      processRunner.mockProcessesForExecutable['flutter'] = <io.Process>[
+        MockProcess.failing() // flutter packages get
+      ];
+
+      Error? commandError;
+      final List<String> output = await runCapturingPrint(
+          runner, <String>['build-examples', '--ios'], errorHandler: (Error e) {
+        commandError = e;
+      });
+
+      expect(commandError, isA<ToolExit>());
+      expect(
+          output,
+          containsAllInOrder(<Matcher>[
+            contains('The following packages had errors:'),
+            contains('  plugin:\n'
+                '    plugin/example (iOS)'),
+          ]));
     });
 
     test('building for iOS when plugin is not set up for iOS results in no-op',
         () async {
-      createFakePlugin('plugin', packagesDir,
-          extraFiles: <String>['example/test']);
```

Comment:
These `extraFiles` changes are unrelated opportunistic cleanup; while copying an existing test in this file I noticed this argument which didn't seem like it should be necessary, and sure enough every test in this file was adding extra test files for no reason.